autogluon_zeroshot/simulation/tabular_predictions.py

```diff
@@ -35,22 +45,229 @@ def predict(self, dataset: str, fold: int, splits: List[str] = None, models: Lis
         assert models is None or len(models) > 0
         return self._predict(dataset, fold, splits, models)
 
-    def models_available_in_dataset(self, dataset: str, fold: int) -> List[str]:
-        """:returns the models available on both validation and test splits"""
-        raise NotImplementedError()
+    def print_summary(self):
+        folds = self.folds
+        datasets = self.datasets
+        tasks = self.tasks
+        models = self.models
+
+        num_folds = len(folds)
+        num_datasets = len(datasets)
+        num_tasks = len(tasks)
+        num_models = len(models)
+
+        folds_dense = self.get_folds_dense()
+        models_dense = self.get_models_dense()
+
+        num_folds_dense = len(folds_dense)
+        num_models_dense = len(models_dense)
+
+        is_dense = self.is_dense()
+        is_dense_folds = self.is_dense_folds()
+        is_dense_models = self.is_dense_models()
+
+        print(f'Summary of {self.__class__.__name__}:\n'
+              f'\tdatasets={num_datasets}\t| folds={num_folds} (dense={num_folds_dense})\t| tasks={num_tasks}\t'
+              f'| models={num_models} (dense={num_models_dense})\n'
+              f'\tis_dense={is_dense} | is_dense_folds={is_dense_folds} | is_dense_models={is_dense_models}')
 
     @property
     def models(self) -> List[str]:
+        """
+        :return: list of models present in at least one dataset.
+        """
+        raise NotImplementedError()
+
+    def get_models(self, present_in_all=False) -> List[str]:
```

Comment:
As discussed offline, lets merge it as the changes are becoming quite significant. I will work on separating most of those methods as free functions into a densifier util.